### ISSUE TITLE
ltrim(class-string, '\\') should return a class-string

### ISSUE THREAD

**Issue by Tofandel** [vimeo/psalm#8218]
When using

```php
    /**
     * Removes the prepended backslash of a class string to conform with how php outputs class names
     *
     * @psalm-param class-string $className
     *
     * @psalm-return class-string
     */
    function normalizeClassName(string $className): string
    {
        return ltrim($className, '\\');
    }
```

2 errors will be thrown
```
ERROR: MoreSpecificReturnType - The declared return type 'class-string' for normalizeClassName is more specific than the inferred return type 'string' (see https://psalm.dev/070)
     * @psalm-return class-string

ERROR: LessSpecificReturnStatement - The type 'string' is more general than the declared return type 'class-string' for normalizeClassName (see https://psalm.dev/129)
        return ltrim($className, '\\');
```

It would be a nice addition to avoid this false negative as this specific ltrim of a class string will always result in the same class string being returned

This issue also exists for phpstan https://github.com/phpstan/phpstan/issues/7483

**Comment by Tofandel** [comment]
https://psalm.dev/r/b6ecdc9b09

**Comment by Tofandel** [comment]
I have very little knowledge of the inner working of psalm, but would https://github.com/vimeo/psalm/compare/4.x...Tofandel:psalm:patch-1
do the trick?
